libs/vkd3d/command.c

```diff
@@ -2607,15 +2602,26 @@ static bool vk_write_descriptor_set_from_d3d12_desc(VkWriteDescriptorSet *vk_des
         case VKD3D_DESCRIPTOR_MAGIC_UAV:
             /* We use separate bindings for buffer and texture SRVs/UAVs.
              * See d3d12_root_signature_init(). */
-            vk_descriptor_write->dstBinding = vk_binding + 2 * index;
+            vk_descriptor_write->dstBinding = vk_binding + (is_uav ? 3 : 2) * index;
             if (descriptor->vk_descriptor_type != VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER
                     && descriptor->vk_descriptor_type != VK_DESCRIPTOR_TYPE_STORAGE_TEXEL_BUFFER)
                 ++vk_descriptor_write->dstBinding;
 
             if (descriptor->vk_descriptor_type == VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER
                     || descriptor->vk_descriptor_type == VK_DESCRIPTOR_TYPE_STORAGE_TEXEL_BUFFER)
             {
-                vk_descriptor_write->pTexelBufferView = &view->u.vk_buffer_view;
+                if (uav_counter)
+                {
+                    vk_descriptor_write->dstBinding += 2;
+                    vk_descriptor_write->pTexelBufferView = &view->vk_counter_view;
+
+                    if (!view->vk_counter_view)
+                        return false;
```

Comment:
Is it okay to not write a descriptor here? Without PARTIALLY_BOUND_BIT we probably have to bind something here. Although this might be somewhat moot since we'll need to do PARTIALLY_BOUND and deferred updates for RS 1.0 soon anyways.

Reply:
It should be fine as long as it isn't statically used by the pipeline, which would only be the case if the app does not provide a counter in a situation where it is required to do so.

Reply:
Right, I think I'm confusing things, from spec:
"• Descriptors that are statically used must have been populated before the descriptor set is
consumed.
• Entries that are not used by a pipeline can have undefined descriptors."

PARTIALLY_BOUND would be for bindless only in that case.